src/parsers/vc12DXMLLoad.cpp

```diff
@@ -19,7 +19,29 @@
 #define LOG_WARNING(...) {}
 #define LOG_INFO(...) {}
 
-// TODO support 'light' and 'dark' variants of all colours
+static bool vc12DXML_IsGreyColour(const char *pStr, uint32_t &out)
+{
+  char *pTemp = udStrdup(pStr);
+  char *tokens[2] = {};
+  int nTokens = udStrTokenSplit(pTemp, " ", tokens, 2);
+  bool success = false;
+
+  if (nTokens == 2 && udStrcmpi(tokens[0], "grey") == 0)
+  {
+    uint32_t scale = udStrAtou(tokens[1]);
```

Comment:
You can avoid the `udStrdup` and `udStrTokenSplit` above by using `udStrBeginsWithi(pStr, "grey ")` and then here `udStrAtou(pStr + 5); // Skip "grey "`. This probably doesn't need to change, but I think it can be simplified.